src/admin_upload/views.py

```diff
@@ -0,0 +1,89 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse, JsonResponse
+from django.contrib.auth.decorators import user_passes_test
+from geonode.layers.models import Dataset
+import json
+import psycopg2
+import sys
+from woolpert import settings
+
+def admin_required(function):
+    def wrap(request, *args, **kwargs):
+        user = request.user
+        if user.is_superuser:
+            return function(request, *args, *kwargs)
+        else:
+            return redirect("/")
+    wrap.__doc__ = function.__doc__
+    wrap.__name__ = function.__name__
+    return wrap
+
+# Create your views here.
+@admin_required
+def admin_form(request):
+    if(request.POST):
+        json_data = json.loads(request.POST.get('json_data'))
+        x_index = [i for i, x in enumerate(json_data["columns"]) if x == 'x_dd']
+        y_index = [i for i, x in enumerate(json_data["columns"]) if x == 'y_dd']
+
+        dbHost     = settings.DATABASE_HOST
+        dbName  = settings.GEONODE_GEODATABASE
+        dbPassword = settings.GEONODE_GEODATABASE_PASSWORD
+        dbUsername    = settings.GEONODE_GEODATABASE_USER
+
+        error_list = []
+
+        conn = psycopg2.connect(
+            database=dbName, user=dbUsername, password=dbPassword, host=dbHost, port="5432"
```

Comment:
Also add `sslmode='${SSL_MODE}'` for secure connections where SSL_MODE will be set from the env variable